Iss 14/profile page

russ430 opened the pull request:
Created the Profile component, front/back end routes.
Profile will request data from back end using the userId from route param.
Function will check whether the current profile matches the logged in user's profile if there is a user logged in.
Back end user request is a public route and retrieves all user data except password.
Profile will conditionally render based on current user's data.

russ430 committed:
Added images

russ430 committed:
Created reusable Button component

russ430 committed:
Created reusable TextBubble component

russ430 committed:
Created Project component and dummy data

russ430 committed:
Created Profile component

russ430 committed:
Updated routes to new path using userId

russ430 committed:
Deleted unused ping file

russ430 committed:
Created Profile component

Profile will conditionally render based on whether current profile is user's own profile

russ430 committed:
Created backend routes to retrieve user data

russ430 committed:
Added check to see if there is a logged in user

skassam21 commented on client/src/App.js:
```diff
@@ -1,61 +1,61 @@
+import Login from "./pages/Login";
+import ProtectedRoutes from "./routes/ProtectedRoutes";
+import { userState } from "./provider/UserContext";
+import { LOADING_USER,UNAUTH_USER } from "./provider/constants";
```
there should be a space between LOADING_USER and UNAUTH_USER - is prettier/eslint set up?

skassam21 commented on client/src/App.js:
```diff
@@ -1,61 +1,61 @@
+  
+  useEffect(() => {
+    const getAuthenticatedUser = async () => {
+        const result = await fetch('/getuser', {
```
this route naming convention (back-end) should be more RESTful

russ430 replied:
Ok. I'm not sure why Github is showing all of these changes as mine but I've only modified the routes for users and launch in this file on lines 51-52. What do you mean as more RESTful though?

skassam21 commented on client/src/pages/Login.js:
```diff
@@ -1,26 +1,26 @@
+  } = useContext(userState);
+  if (isAuthenticated) {
+    history.push(`/users/${user._id}`);
+  }
```
I don't love that this logic is in this route, but I'm okay with it

skassam21 commented on client/src/pages/Profile.js:
```diff
@@ -1,13 +1,214 @@
+    };
+    getUser();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [userId]);
```
We actually shouldn't have to get the user again if you are on the page of the current logged in user (since we have the user from the context)

skassam21 reviewed (commented):
Looks good! Excited to see what this looks like in standup

russ430 committed:
Ran Prettier through code

russ430 committed:
Updated per Shum's comments

Removed extra call to back end for user data if it's the currently logged in user (we have that data in Context)

russ430 committed:
Updated NavBar Profile button to correct nav path
